src/openstudio_lib/MainWindow.cpp

```diff
@@ -303,10 +317,41 @@ bool MainWindow::verboseOutput() const {
   return m_verboseOutput;
 }
 
+void MainWindow::onVerticalTabSelected(int verticalTabId) {
+  if (allowAnalytics()) {
+    emit sendAnalytics(m_analyticsId, verticalTabId);
+  }
+}
+
 void MainWindow::toggleVerboseOutput(bool verboseOutput) {
   m_verboseOutput = verboseOutput;
 }
 
+void MainWindow::promptAnalytics() {
+  if (m_analyticsId.isEmpty()) {
+    QMessageBox::StandardButton reply;
+    // TODO: link to a privacy page on the website, explain why we need metrics
+    reply = QMessageBox::question(this, tr("Allow Analytics"),
+                                  tr("Allow OpenStudio Coalition to collect anonymous usage statistics to help improve the OpenStudio Application?"),
+                                  QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
+    if (reply == QMessageBox::Yes) {
+      emit enableAnalytics(true);
+    } else {
+      emit enableAnalytics(false);
+    }
+  }
+}
```

Comment:
:+1: Confirmed that we do get the prompt only once, until the settings are written